### ISSUE TITLE
DefaultAzureCredential authentication failed

### ISSUE THREAD

**Issue by HametAk** [Azure/terraform-provider-azapi#144]
I have a big issue with a module which uses AzApi.

I get the following error:
It is not possible for me to do a terraform apply with AzApi.

I already tried to set environment variables manually and only do az login on specific tenant.
This is reproducible in my company.

The environment used is the Azure Virtual Desktop and error only occurs there.
![MicrosoftTeams-image](https://user-images.githubusercontent.com/70479098/177104305-468a9238-4cb1-4fbc-b752-07ad9c3d5cba.png)

**Comment by ms-henglu** [comment]
Hi @HametAk ,

It seems that you didn't configure the credentials properly, please check https://registry.terraform.io/providers/Azure/azapi/latest/docs/guides/serivce_principal_client_sercret and see if there're some fields are missed?

**Comment by HametAk** [comment]
Hi @ms-henglu ,
thanks for quick reply.

Is this necessary? Because I am not using a service principal or is it mandatory/recommended to use one?

**Comment by ms-henglu** [comment]
Hi @HametAk - No, you can check the [guides](https://registry.terraform.io/providers/Azure/azapi/latest/docs) on the left side, there're other ways to authenticate with azure.

**Comment by HametAk** [comment]
I am just using simple "az login -t <tenant>" on my other environment which works just fine.
It's just an issue when using the Azure Cloud Desktop.
I can do every other terraform task except AzAPI

**Comment by ms-henglu** [comment]
@HametAk

> I can do every other terraform task except AzAPI

You mean, `azurerm` provider works fine?

**Comment by HametAk** [comment]
@ms-henglu 
Yes it works fine.

**Comment by HametAk** [comment]
Following Code was used:
```

terraform {
  required_providers {
    azapi = {
      source  = "Azure/azapi"
      version = "~> 0.3.0"
    }
  }
}

resource "azapi_update_resource" "subscription_tags" {
  type      = "Microsoft.Resources/tags@2021-04-01"
  parent_id = "/subscriptions/${var.subscription_id}"
  name      = "default"
  body = jsonencode({
    properties = {
      tags = var.tags
    }
  })
}
```

**Comment by andrewPoland** [comment]
I ran into this issue too when running from an Azure hosted VM. The VM has no System or User Assigned Identities and I'm attempting to use `az login` for authentication. 

Reverting back to version `0.2.0` allowed me to create the resources using credentials from `az login`.  

It seems like it's related to the VM having no Identities assigned to it based on the 400 error message. 

Downgrading the version isn't ideal, is there any way to disable the attempt to use user assigned identities via environment variables or configuration?

**Comment by ms-henglu** [comment]
link to https://github.com/Azure/azure-sdk-for-go/issues/18352

**Comment by ms-henglu** [comment]
Hi all,

The bug fix has been released in azapi 0.5.0, would you please give it a try?
